src/test/java/nuffle/parserTest.java

```diff
@@ -0,0 +1,29 @@
+package nuffle;
+
+import nuffle.parser.Parser;
+import nuffle.task.Deadline;
+import nuffle.task.Task;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class parserTest {
```

Comment:
perhaps you could create tests for strings that are not formatted properly, e.g. missing deadline, missing spaces between some words, etc. to see if your `Parser` class can correctly catch such errors and either throw a checked exception or reject the user's input, instead of potentially crashing the entire program